libflux/src/ast/walk/mod.rs

```diff
@@ -224,29 +224,36 @@ impl<'a> Node<'a> {
     }
 }
 
-// Visitor defines a visitor pattern for walking the AST
-//
-// When used with the walk function, Visit will be called for every node
-// in depth-first order. After all children for a Node have been visted,
-// Done is called on that Node to signal that we are done with that Node.
-//
-// If Visit returns None, walk will not recurse on the children.
-//
-// Note: the Rc in visit and done is to allow for multiple ownership of a node, i.e.
-//       a visitor can own a node as well as the walk funciton. This allows
-//       for nodes to persist outside the scope of the walk function and to
-//       be cleaned up once all owners have let go of the reference.
+/// Visitor defines a visitor pattern for walking the AST.
+///
+/// When used with the walk function, Visit will be called for every node
+/// in depth-first order. After all children for a Node have been visted,
+/// Done is called on that Node to signal that we are done with that Node.
+///
+/// If Visit returns None, walk will not recurse on the children.
+///
+/// Note: the Rc in visit and done is to allow for multiple ownership of a node, i.e.
+///       a visitor can own a node as well as the walk funciton. This allows
+///       for nodes to persist outside the scope of the walk function and to
+///       be cleaned up once all owners have let go of the reference.
+///
+/// Implementors of the Visitor trait will typically wrap themselves in Rc and RefCell
```

Comment:
I converted this to a doc comment so that it will be rendered by `cargo doc`, and also added this section here.